packages/core/src/formula/formula.ts

```diff
@@ -142,7 +142,23 @@ export type ToddleEnv =
     }
 
 export function isFormula(f: any): f is Formula {
-  return f && typeof f === 'object' && typeof f.type === 'string'
+  return (
+    f &&
+    typeof f === 'object' &&
+    typeof f.type === 'string' &&
+    [
+      'path',
+      'function',
+      'record',
+      'object',
+      'array',
+      'or',
+      'and',
+      'apply',
+      'value',
+      'switch',
+    ].includes(f.type)
```

Comment:
Just to make sure we don't include something that's not a valid type for a formula, consider this type casting
```suggestion
    ([
      'path',
      'function',
      'record',
      'object',
      'array',
      'or',
      'and',
      'apply',
      'value',
      'switch',
    ] as Formula['type'][]).includes(f.type)
```

Reply:
Yes! Great idea.